src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * to represent commonalities from Phone, Email and Address
+ */
+
+public class Contact {
+    
+    public final String value;
+    public boolean isPrivate;
+    
+    public Contact(String contact, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        this.value = contact;
+    }
+    
+    public String toString() {
```

Comment:
I guess you should mark these methods are overridden methods.
From [Java documentation](https://docs.oracle.com/javase/tutorial/java/annotations/predefined.html):

> While it is not required to use this annotation when overriding a method, it helps to prevent errors. If a method marked with @Override fails to correctly override a method in one of its superclasses, the compiler generates an error.